File: Assignment37/Java/Program3.java
// PS - write a program which accepts 2 strings from user and check whether first N content of 2 strings are equal or not

import java.util.*;

class StringX
{
    public boolean StringComp(String str1, String str2, int No)
    {
        str1 = str1.toLowerCase();
        str2 = str2.toLowerCase();

        char Arr[] = str1.toCharArray();
        char Brr[] = str2.toCharArray();

        int i = 0;
        for(i = 0; i < No; i++)
        {
            if(Arr[i] != Brr[i])
            {
                return false;
            }
        }
        return true;
    }
}

class Program3
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter 1st string");
        String str1 = sobj.nextLine();

        System.out.println("Enter 2nd string");
        String str2 = sobj.nextLine();

        System.out.println("Enter number of characters that you want to compare");
        int Value = sobj.nextInt();

        StringX obj = new StringX();

        boolean bret = obj.StringComp(str1, str2, Value);

        if(bret == true)
        {
            System.out.println("The first "+ Value + "number of characters from string are same");
        }
        else
        {
            System.out.println("The first "+ Value + "number of characters from string are not same");
        }
    }
}

File: Assignment54/C/Program3.c
// PS - Write a recursive program which accepts string from user and return length
// hello
// 5

#include <stdio.h>
#include <stdlib.h>

int DisplayR(char *arr)
{
    static int iSum = 0;
    if(*arr != '\0')
    {
        iSum++;
        arr++;
        DisplayR(arr);
    }
    return iSum;
}

int main()
{
    char str[20];
    int iRet = 0;

    printf("Enter string \n");
    scanf("%[^'\n']s", str);

    iRet = DisplayR(str);
    printf("Count of characters are %d \n", iRet);

    return 0;
}